### PR TITLE
Add ItemArmor#onArmorTick

### PR DESCRIPTION
**Pull request by shadowfacts**
Called every tick while the armor is being worn. Allows for items such as [sleeping bags](https://github.com/shadowfacts/SleepingBag/blob/1.9.4/src/main/java/net/shadowfacts/sleepingbag/ItemSleepingBag.java#L103-L132).

### CODE REVIEW

File: patches/net/minecraft/entity/player/InventoryPlayer.java.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
+--- /net/minecraft/entity/player
++++ /net/minecraft/entity/player
+@@ -270,6 +270,11 @@
+             }
+         }
+ 
++        for (ItemStack armor : this.armorInventory) {
++            if (armor != null && armor.getItem() instanceof ItemArmor) {
++                ((ItemArmor)armor.getItem()).armorWornTick(player, armor);
```
Comment (Thog) [inline patches/net/minecraft/entity/player/InventoryPlayer.java.patch]: Can you remove those brackets? They are unneeded

**Comment by shadowfacts** [comment]
The `EntityUpdate` event will be used instead.

**Comment by darkhax** [comment]
@shadowfacts Why did you open this back up?

**Comment by darkhax** [comment]
To add from the previous discussions on Discord, it has been agreed that this would be a good feature. A new event will be created, along side the new method.

**Comment by shadowfacts** [comment]
Updated to include `UpdateEvent.ArmorUpdate`.

File: patches/net/minecraft/entity/player/InventoryPlayer.java.patch [hunk 2]
```diff
@@ -0,0 +1,14 @@
+--- /net/minecraft/entity/player
++++ /net/minecraft/entity/player
+@@ -270,6 +270,11 @@
+             }
+         }
+ 
++        for (ItemStack armor : this.armorInventory)
++            if (armor != null) {
++                xyz.openmodloader.event.impl.UpdateEvent.ArmorUpdate.handle(player, armor);
++                armor.getItem().onArmorTick(player, armor);
```
Comment (lclc98) [inline patches/net/minecraft/entity/player/InventoryPlayer.java.patch]: This should probably done in the handle method

File: src/main/java/xyz/openmodloader/event/impl/UpdateEvent.java [hunk 3]
```diff
@@ -131,4 +134,47 @@ public MinecraftServer getServer() {
...
+     */
+    public static class ArmorUpdate extends UpdateEvent {
+
+        /**
+         * The player whose armor is being ticked.
+         *
+         * @see #getPlayer()
+         */
+        private final EntityPlayer player;
+        /**
```
Comment (darkhax) [inline src/main/java/xyz/openmodloader/event/impl/UpdateEvent.java]: New line added here

File: src/main/java/xyz/openmodloader/event/impl/UpdateEvent.java [hunk 4]
```diff
@@ -131,4 +134,47 @@ public MinecraftServer getServer() {
...
+         * @see #getStack()
+         */
+        private final ItemStack stack;
+
+        private ArmorUpdate(EntityPlayer player, ItemStack stack) {
+            this.player = player;
+            this.stack = stack;
+        }
+
+        /**
```
Comment (darkhax) [inline src/main/java/xyz/openmodloader/event/impl/UpdateEvent.java]: I know it's redundant, but we need body documentation as well. This applies to all other methods.

File: src/main/java/xyz/openmodloader/event/impl/UpdateEvent.java [hunk 5]
```diff
@@ -131,4 +134,52 @@ public MinecraftServer getServer() {
...
+         * Gets the armor stack that is being ticked.
+         *
+         * @return The stack that contains the armor
+         */
+        public ItemStack getStack() {
+            return stack;
+        }
+
+        public static void handle(EntityPlayer player, ItemStack stack) {
```
Comment (darkhax) [inline src/main/java/xyz/openmodloader/event/impl/UpdateEvent.java]: Handle also needs docs :p

**Pull request merged**